dockstore-webservice/src/main/java/io/dockstore/webservice/core/Notebook.java

```diff
@@ -72,8 +72,11 @@ public EntryTypeMetadata getEntryTypeMetadata() {
     }
 
     @Override
-    public void setParentEntry(Entry parentEntry) {
-        throw new UnsupportedOperationException("cannot add a checker workflow to a Notebook");
+    public void setParentEntry(Entry<?, ?> parentEntry) {
+        if (parentEntry == null) {
+            return;
+        }
+        throw new UnsupportedOperationException("Notebook cannot be a checker workflow");
```

Comment:
Is this message reversed?  In the  old implementation,  `parentEntry` is the checker workflow being added  and we don't check  what type it is.

Reply:
`setParentEntry` is used to set the parent of a checker workflow, so when it's invoked with a non-null argument and `this` is a notebook, we throw with this message because a notebook can't be a checker workflow.  The previous implementation was incorrect, it implemented `setParentEntry` and `setIsChecker` as if their purpose was to add a checker to the notebook/service, but their actual purpose is to mark a notebook/service as a checker and set its parent (the checked workflow).